File: lib/domain/shared/ports/cultivation_plan_phase_broadcast_port.rb
# frozen_string_literal: true

module Domain
  module Shared
    module Ports
      module CultivationPlanPhaseBroadcastPort
        def broadcast_phase_update(plan_id:, channel_class:, payload:)
          raise NotImplementedError, "#{self.class}#broadcast_phase_update"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/shared/ports/cultivation_plan_phase_broadcast_port.rs
use serde_json::Value;

/// Ruby: `Domain::Shared::Ports::CultivationPlanPhaseBroadcastPort`
pub trait CultivationPlanPhaseBroadcastPort: Send + Sync {
    fn broadcast_phase_update(
        &self,
        plan_id: i64,
        channel_class: &str,
        payload: &Value,
    );
}
